test/widgets/content_test.dart

```diff
@@ -89,6 +94,97 @@ void main() {
     });
   });
 
+  group('Spoiler', () {
+    testContentSmoke(ContentExample.spoilerDefaultHeader);
+    testContentSmoke(ContentExample.spoilerPlainCustomHeader);
+    testContentSmoke(ContentExample.spoilerRichHeaderAndContent);
+
+    group('interactions: spoiler with tappable content (an image) in the header', () {
+      Future<List<Route<dynamic>>> prepareContent(WidgetTester tester, String html) async {
+        addTearDown(testBinding.reset);
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+        prepareBoringImageHttpClient();
+
+        final pushedRoutes = <Route<dynamic>>[];
+        final testNavObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+        await tester.pumpWidget(GlobalStoreWidget(child: MaterialApp(
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          navigatorObservers: [testNavObserver],
+          home: PerAccountStoreWidget(accountId: eg.selfAccount.id,
+            child: MessageContent(
+              message: eg.streamMessage(content: html),
+              content: parseContent(html))))));
+        await tester.pump(); // global store
+        await tester.pump(); // per-account store
+        debugNetworkImageHttpClientProvider = null;
+
+        // `tester.pumpWidget` introduces an initial route;
+        // remove it so consumers only have newly pushed routes.
+        assert(pushedRoutes.length == 1);
+        pushedRoutes.removeLast();
+        return pushedRoutes;
+      }
+
+      void checkIsExpanded(WidgetTester tester,
+        bool expected, {
+        Finder? contentFinder,
+      }) {
+        final sizeTransition = tester.widget<SizeTransition>(find.ancestor(
+          of: contentFinder ?? find.text('hello world'),
+          matching: find.byType(SizeTransition),
+        ));
+        check(sizeTransition.sizeFactor)
+          ..value.equals(expected ? 1 : 0)
+          ..status.equals(expected ? AnimationStatus.completed : AnimationStatus.dismissed);
+      }
+
+      const example = ContentExample.spoilerHeaderHasImage;
+
+      testWidgets('tap image', (tester) async {
+        final pushedRoutes = await prepareContent(tester, example.html);
+
+        await tester.tapAt(tester.getCenter(find.byType(RealmContentNetworkImage)));
```

Comment:
nit: can simplify to:
```suggestion
        await tester.tap(find.byType(RealmContentNetworkImage));
```